app/Http/Controllers/V2/AuditStatus/StoreAuditStatusController.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Http\Controllers\V2\AuditStatus;
+
+use App\Http\Controllers\Controller;
+use App\Http\Resources\V2\AuditStatusResource;
+use App\Models\Traits\SaveAuditStatusTrait;
+use App\Models\V2\AuditAttachment\AuditAttachment;
+use App\Models\V2\AuditStatus\AuditStatus;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Storage;
+
+class StoreAuditStatusController extends Controller
+{
+    use SaveAuditStatusTrait;
+
+    public function __invoke(Request $request): AuditStatusResource
+    {
```

Comment:
This method will also benefit from the `ModelBindingInterfaceMiddelware` treatment mentioned above. With that, the `Auditable` will come in as a bound parameter to the method, as above. That means that `type` will no longer be sent in the body of the request, but will be in the URL along with the UUID.

Then this method gets a bit simpler as well. Note that I think the `type` member on `AuditStatus` becomes superfluous with these changes. Unless I'm missing something, the type of an audit is just the kind of model it's associated with.

This method should be able to be something more like:
```
  if ($auditable instanceof UpdateRequest) {
    $auditable->auditStatuses()->where('is_active', true)->update('is_active', false);
    $auditStatus = $this->saveAuditStatus($auditable, $body['status'], $body['comment'], $body['is_active'], $body['request_removed']; 
  } else {
    $auditStatus = $this->saveAuditStatus($auditable, $body['status'], $body['comment']);
  }
```

The file saving below is fine as-is I think, although see my comments on AuditAttachment; this would likely be better handled with Spatie. The `$auditStatusresponse` variable would be better named `$auditStatus` though, as that's what it is under the hood.

Reply:
There is another `Auditable` already, so I went for `AuditableModel` following `EntityModel`.
I just couldn't make the store method to work with `ModelInterfaceBindingMiddleware`.

But I think this is good enough improvement.